core-api/src/main/java/com/optimizely/ab/bucketing/DecisionService.java

```diff
@@ -130,8 +130,8 @@ public DecisionService(@Nonnull Bucketer bucketer,
 
         if (ExperimentUtils.isUserInExperiment(projectConfig, experiment, filteredAttributes)) {
             String bucketingId = userId;
-            if (filteredAttributes.containsKey(BUCKETING_ATTRIBUTE)) {
-                bucketingId = filteredAttributes.get(BUCKETING_ATTRIBUTE);
+            if (filteredAttributes.containsKey(ReservedAttributeKey.BUCKETING_ATTRIBUTE.toString())) {
```

Comment:
why you need to add toString, can u plz. check without toString()?
@mnoman09

Reply:
No for enums key in java there is no simpler way than that. I can use constants instead of enums.